tests/test-util/test.manifest.js

```diff
@@ -35,6 +38,15 @@ describe('util/manifest', () => {
         })
     ));
 
+    it('allows manifests without IDs', () => withTempDir(
```

Comment:
It's different in that one tests a manifest without the `applications` property at all and the other tests for an `applications.gecko` property but with an undefined `applications.gecko.id` property. I updated one of the test titles to make this more clear.